src/main/java/org/springframework/guice/module/GuiceAutowireCandidateResolver.java

```diff
@@ -81,6 +82,9 @@ protected boolean isLazy(DependencyDescriptor descriptor, String beanName) {
 			beanFactory.doResolveDependency(descriptor, beanName, null, null);
 		}
 		catch (NoSuchBeanDefinitionException ex) {
+			if (ex instanceof NoUniqueBeanDefinitionException) {
+				throw ex;
```

Comment:
A proxy being swapped in caused these bugs to go unnoticed until someone started using the bean and found duplicated.